src/main/java/command/AddToDoCommand.java

```diff
@@ -0,0 +1,71 @@
+package command;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import duke.Storage;
+import duke.TaskList;
+import exception.DukeException;
+import exception.EmptyTodoException;
+import task.Task;
+import task.ToDo;
+import ui.Ui;
+
+/**
+ * Represents a <code>Command</code> whose task is adding a <code>ToDo</code> to the <code>TaskList</code>.
+ * The <code>AddToDoCommand</code> object contains an array of <code>String</code> which is an array
+ * containing a command and the argument of the command (if any).
+ */
+public class AddToDoCommand extends Command {
+
+    public AddToDoCommand(String[] splitCommand) {
+        super(splitCommand);
+    }
+
+    /**
+     * Adds the <code>ToDo</code> to <code>tasks</code> and saves the <code>ToDo</code> to
+     * <code>storage</code>.
+     *
+     * @param tasks  Task list of the Duke.
+     * @param ui Ui of the Duke.
+     * @param storage Storage of the Duke.
+     * @return The response text.
+     * @throws DukeException If failed to save to <code>storage</code> or no description provided in
+     * <code>splitCommand</code>.
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        try {
+            String description = splitCommand[1];
+            Task toAdd = new ToDo(description);
+
+            tasks.add(toAdd);
+            storage.save(tasks);
```

Comment:
I realised that you chose to save the file to storage after execution of each command. I was wondering if that was indeed necessary, given that file I/O operations are in general more expensive and might slow down the functionalities. Instead, you could try using storage to write out the file after multiple commands are executed only at the end of duke's life cycle when the bye command is issued?

This is understandably more of a design decision though.

Reply:
That's right, but the trade-off is that, if the user closes the program (without the bye command) or if the program is closed unexpectedly, the unsaved tasks will not be written.
Anyway, feel free to reply to this comment if you have further clarification/suggestion regarding this issue!